unittests/integration/ghz_nisq_tester.cpp

```diff
@@ -38,16 +38,24 @@ CUDAQ_TEST(GHZSampleTester, checkSimple) {
     EXPECT_TRUE(bits == "00000" || bits == "11111");
   }
   EXPECT_EQ(counter, 1000);
-  printf("Exp: %lf\n", counts.exp_val_z());
+  printf("Exp: %.16lf\n", counts.exp_val_z());
 }
 
 CUDAQ_TEST(GHZSampleTester, checkBroadcast) {
 
+  // FIXME Issue-608 - I don't think this is enough to make it fully repeatable
+  // yet, even if you set OMP_NUM_THREADS=1
```

Comment:
I looked into this briefly today. This is because `qpp` creates `thread_local` random number generator instances. In the broadcast mode, we execute those quantum tasks on a different thread hence the `set_random_seed` call here on the `main` thread has no effect. 

This can be verified by adding `#define NO_THREAD_LOCAL_` before `#include "qpp.h"` in `QppCircuitSimulator.cpp`; then the results will be repeatable. I'm not suggesting disabling `thread_local` in `qpp` though since it could have other perf. implications. Maybe a more generic solution would be setting the seed value on the `sample` call itself...

Reply:
Thanks, Thien. That's a nice find. I'll add this note to Issue #608.